### PR TITLE
Fix hot reload

### PR DESCRIPTION
**Pull request by chriddyp**
Fixes https://github.com/plotly/dash/issues/634

### CODE REVIEW

**Comment by chriddyp** [comment]
hotreload

**Comment by chriddyp** [comment]
It looks like our existing hot reload tests don't test python changes yet (only css changes).

I think that getting proper python integration tests here might be a bit more work, so I'd like to get this fix merged in first and then loop back later on integration tests.

**Comment by chriddyp** [comment]
Best to review this one commit-by-commit

File: src/components/core/Reloader.react.js [hunk 1]
```diff
@@ -90,29 +125,23 @@ class Reloader extends React.Component {
...
-                    window.clearInterval(this.state.intervalId);
+                    // Backend code changed - can do a soft reload in place
                     dispatch({type: 'RELOAD'});
                 }
             }
         } else if (reloadRequest.status === 500) {
             if (this._retry > this.state.max_retry) {
-                window.clearInterval(this.state.intervalId);
+                this.clearInterval();
                 // Integrate with dev tools ui?!
```
Comment (byronz) [inline src/components/core/Reloader.react.js]: was wondering if we should throw and the error and see the comment. yes!

File: src/components/core/Reloader.react.js [hunk 2]
```diff
@@ -124,16 +174,20 @@ class Reloader extends React.Component {
         const {dispatch} = this.props;
         const {disabled, interval} = this.state;
         if (!disabled && !this.state.intervalId) {
-            const intervalId = setInterval(() => {
-                dispatch(getReloadHash());
+            const intervalId = window.setInterval(() => {
+                // Prevent requests from piling up - reloading can take
```
Comment (byronz) [inline src/components/core/Reloader.react.js]: :cat:

**Comment by byronz** [comment]
my current understanding of this part leading to a black box reviewing, I verified the fix with the reloading issue I see with dash-docs, from the network traffic, it seems works properly now with both hard and soft reload.  :dancer:  

better have another approval from @alexcjohnson  or @Marc-Andre-Rivet

**Pull request merged**